### DISCUSSION TITLE
Helix cant find themes

### DISCUSSION THREAD

**Discussion by Dorfeuheinz** [helix-editor/helix#7549] [Q&A]
I installed helix editor the other day & I cant change my themes. When using the `:theme` command the only thing thats displayed is default & base16_default. I find no other themes in my editor. whats wrong?

**Comment by aromeronavia** [comment]
CAn you give more context on your setup?
- Operating System
- How did you install Helix (brew/apt-get/source-code)

**Reply by Dorfeuheinz** [reply]
OS- deepin linux
Installation- Downloaded from source & built it. apt-get doesnt work on debian apparently

**Comment by pascalkuthe** [comment]
Your runtime folder is missing, make sure you.followed the install instructions correctly

**Reply by Dorfeuheinz** [reply]
i do have a runtime folder. It has all the themes files in it yet idk why only 2 themes are available for usage. I installed it in the past, didnt have a problem with it. Idk whys this showing up rn. Is there any command to recompile helix?

**Reply by the-mikedavis** [reply]
The themes aren't compiled in so rebuilding shouldn't have any effect. You should have a file `~/.config/helix/runtime/themes/onedark.toml` or have a `$HELIX_RUNTIME` environment variable that points to the `runtime` directory in the source repository.

**Reply by Dorfeuheinz** [reply]
Nope this doesnt work too, tried both of those methods. Somethings still missing. The .config/helix option doesnt exist, I created those dirs, & then a symlink to the runtime but nope, didnt work

**Reply by Dorfeuheinz** [reply]
I actually fixed that by changing base16 theme to the 1 I needed & recompiling but the autocomplete still doesnt work. My lan servers work well but the auto suggestions arent working, idk things r broken rn

**Reply by CptPotato** [reply]
> Nope this doesnt work too, tried both of those methods. Somethings still missing. The .config/helix option doesnt exist, I created those dirs, & then a symlink to the runtime but nope, didnt work

You can try `hx --health` and check the paths at the start of the output to see where helix tries to look for the runtime files.

**Reply by Dorfeuheinz** [reply]
i found a workaround. changed the default theme to the one I needed, so the name is the same but file contents r different. Lang servers r running fine ig

**Comment by rochacbruno** [comment]
I just fixed by doing the steps:

```bash
git clone https://github.com/helix-editor/helix
cd helix
cargo install --path helix-term --locked
```
THEN THE IMPORTANT PART that is missing from the docs page

```bash
cp -R runtime ~/.config/helix/
hx -g fetch
hx -g build
```

**Reply by the-mikedavis** [reply]
Runtime directory setup is documented right here: https://docs.helix-editor.com/master/install.html#configuring-helixs-runtime-files

**Reply by rochacbruno** [reply]
Sorry, I overlooked that section, it was confusing to me, I tried to set `HELIX_RUNTIME` to `~/.config/helix/runtime` as it was not clear to me that I needed to point it to the repo src folder.

Then I solved by copying from repo to my home config foder, which is equivalente to the symlink option mentioned there.

That was my bad, but anyway this comment here can help someone else that ends up in the same situation.

Thanks

**Comment by MichaelLan** [comment]
I have the same problem, I tried to install directly using apt install in wsl2 (ubuntu), but the themes are not displayed, after I removed it and tried to use the source code and I had the same problem. 

I set `$HELIX_RUNTIME` `/home/michaellan/.config/helix/runtime`  in my zsh config file 

I copy the runtime folder inside `/home/michaellan/.config/helix/runtime`, and the themes folder is there.

With the hx --health command I get
```
Config file: /home/michaellan/.config/helix/config.toml
Language file: default
Log file: /home/michaellan/.cache/helix/helix.log
Runtime directories: /home/michaellan/.config/helix/runtime;/home/michaellan/.config/helix/runtime;/home/michaellan/.config/helix/runtime;/home/michaellan/.cargo/bin/runtime
Clipboard provider: termcode
System clipboard provider: termcode
```
I have no other idea how to solve it.

**Comment by webdev23** [comment]
The env variable `HELIX_RUNTIME=` is not propagated on some machines, not sure why. 
The `ln` command given in the docs doesn't either on those machines.
What worked for me in this situation was:

`rm -rf ~/.config/helix && ln -sf ~/helix/runtime ~/.config/helix`

**Reply by JayeshUkey** [reply]
This is the only solution which worked. TYhanks a Ton.

**Comment by OscarRaizer** [comment]
do not forget to push space after :theme to see another themes

**Reply by NGC717** [reply]
YOU SAVED ME FROM HELL😭. 

The first time I tried Helix in WSL, I was able to select a theme correctly (I must have unconsciously typed :theme [space], and the theme displayed properly). 

For some reason, I later uninstalled WSL, and when I tried again, no matter what I did, I couldn't get the theme to display correctly. I asked AI, none of them helped. I searched every relevant keyword on search engines, watched video tutorials, and was just about to give up when I stumbled upon this comment. 

Suddenly, everything became clear: except for that very first time, every subsequent attempt I made used :theme [Enter], so it only ever showed the "default" theme. 

THANK YOU SO MUCH!
